common/types.go

```diff
@@ -537,6 +541,81 @@ func (l Location) MarshalJSON() ([]byte, error) {
 	return json.Marshal(intSlice)
 }
 
+// NewLocation verifies the inputs for region and zone and returns a valid location
+func NewLocation(region, zone int) (Location, error) {
+	loc := Location{}
+	err := loc.SetRegion(region)
+	if err != nil {
+		return nil, err
+	}
+	err = loc.SetZone(zone)
+	if err != nil {
+		return nil, err
+	}
+
+	return loc, nil
+}
+
+func (l *Location) SetRegion(region int) error {
+	if region < 0 || region >= 0xf {
+		return ErrInvalidLocation
+	}
+	if len(*l) < 1 {
+		// Extend location to include region if its too short
+		newLoc := make([]byte, 1)
+		*l = newLoc
+	}
+	(*l)[0] = byte(region)
+	return nil
+}
+
+func (l *Location) SetZone(zone int) error {
+	if zone < 0 || zone > 0xf {
+		return ErrInvalidLocation
+	}
+	if len(*l) < 2 {
+		// Extend the slice while preserving the first byte, if it exists
+		newSlice := make([]byte, 2)
+		if len(*l) > 0 {
+			newSlice[0] = (*l)[0] // Preserve existing first byte
+		}
+		*l = newSlice
+	}
+	(*l)[1] = byte(zone)
+	return nil
+}
+
+// LocationFromName parses a location name and returns a Location.
+func LocationFromName(name string) (Location, error) {
+	if name == "prime" {
+		return Location{}, nil
+	}
+
+	parts := strings.Fields(name)
+	if len(parts) == 1 {
```

Comment:
I think the name displayed to the user (in their preferred language) need not necessarily be part of consensus. It may make protocol upgrades or forks easier with numbers rather than relying on mental mapping with names.

Reply:
fair, but surely how we encode a name should be the reciprocal of how we decode a name, right? Here you've added a function to decode a name into a location, but it is not able to decode the names output by the encoder method.

essentially, the reflexive property should hold:
`loc == LocationFromName(loc.Name())`

Reply:
I see what you mean now, I'll modify the function to accept either given the ambiguity.